server.js

```diff
@@ -1,11 +1,24 @@
 var express = require("express");
 var mongoose = require("mongoose");
+var bodyParser = require("body-parser");
 var app = express();
 
 app.use("/angular", express.static(__dirname + "/node_modules/angular"));
 app.use("/jquery", express.static(__dirname + "/node_modules/jquery/dist"));
 app.use("/bootstrap", express.static(__dirname + "/node_modules/bootstrap/dist"));
 app.use(express.static(__dirname + "/public"));
+app.use(bodyParser.json());
+
+var contactSchema = mongoose.Schema
```

Comment:
I moved both the schema and model definitions to global scope for use in both endpoints. For one, you can't recompile the model, and two, it's DRY. This entire file needs desperately to be abstracted as well as the front-end stuff.